proposals/052-cryptography-leg-1.md

```diff
@@ -0,0 +1,329 @@
+
+Furthermore, bindings to a dedicated cryptography library may yield a performance benefit, as one would expect a popular C++ crypto library such as Botan to be highly performant. Testing in GHCi with `:set +s` for a requested example of `bcrypt` yielded the following results:
+
+```
+-- Botan
+ghci> bcryptGenerateIO "Fee fi fo fum!" r 16
+"$2a$16$j3Obv.HNT8zQ9OXFXEJeQeq3nT5WmUecPYDF0tc3grNsX9N0uDSCm\NUL"
+(3.73 secs, 996,152 bytes)
+
+-- cryptonite
+ghci> hashPassword 16 ("Fee fi fo fum!" :: ByteString) :: IO ByteString
+"$2b$16$nmHHDe/wnfSl85dIFtqjMuAHxqAmL.JQ6knp8e6kf.X1V9/LCZxUy"
+(8.84 secs, 2,138,266,840 bytes)
+```
+
+It was quick and dirty testing in GHCi so your mileage may vary, but the results indicate that bindings to Botan may be significantly faster and consumes less memory than `crypton/ite`, if other modules / functions are similarly performant. Further implentation, testing, and benchmarking will be necessary to confirm this.
+
+# Prior Art and Related Efforts
+
+We can group Haskell cryptography efforts into several categories, which are not necessarily distinct:
+
+- Libraries for specific algorithms
+- Cryptographic "kitchen sinks"
+- Bindings to foreign libraries
+- Efforts by [Thomas DuBuisson](https://hackage.haskell.org/user/ThomasDuBuisson)
+- Efforts by [Vincent Hanquez](https://hackage.haskell.org/user/VincentHanquez)
+
+The wiki lists [several libraries](https://wiki.haskell.org/Applications_and_libraries/Cryptography) and Hackage lists [many more](https://hackage.haskell.org/packages/#cat:Cryptography) according to the category.
+
+> NOTE: Cryptocurrency libraries and libraries not on Hackage are excluded from consideration.
+
+Specific libraries worth noting are:
+
+- Predecessors to `cryptonite` - all deprecated
+    - [cryptohash](https://hackage.haskell.org/package/cryptohash)
+    - [cryptocipher](https://hackage.haskell.org/package/cryptocipher)
+    - [crypto-pubkey](https://hackage.haskell.org/package/crypto-pubkey)
+    - [crypto-random](https://hackage.haskell.org/package/crypto-random)
+- Cryptographic "kitchen sinks"
+    - [cryptonite](https://hackage.haskell.org/package/cryptonite)
+    - [crypton](https://hackage.haskell.org/package/cryptonite) - a fork of cryptonite
+- Bindings to libsodium
+    - [NaCl](https://github.com/serokell/haskell-crypto)
+    - [saltine](https://hackage.haskell.org/package/saltine)
+    - [sel](https://hackage.haskell.org/package/sel-0.0.1.0/candidate)
+- Bindings to botan
+    - [Z-Botan](https://hackage.haskell.org/package/Z-Botan) - part of Z-Haskell, GHC8
+- Honorable mentions
+    - [Crypto](https://hackage.haskell.org/package/Crypto) - a very old sink
+    - [crypto-api](https://hackage.haskell.org/package/crypto-api) - cryptographic classes
+    - [cryptol](https://hackage.haskell.org/package/cryptol) - a DSL for developing new cryptographic primitives
```

Comment:
I don't think Cryptol is particularly relevant here - it's a language that is not Haskell, but is implemented in Haskell. It's used to specify crypto primitives, but executable Cryptol specs are not realistic parts of Haskell libraries intended fro broad use.

```suggestion
```

Reply:
I included it for completeness, but I can strike its mention here as it was definitely an outlier.